### PR TITLE
Convert Blob Source to HttpUrl when expanding template

### PR DESCRIPTION
**Pull request by bgklein**
Fix missed attribute in new ExtendedResourceFile object

### CODE REVIEW

**Comment by xingwu1** [comment]
I think again, I don't think we need this change. We can ask the user who use our SDK extension to update their code to use httpUrl instead of blobSource. This is kind of different behavior with JSON script user.

**Comment by timotheeguerin** [comment]
I think it would be nice to support it for a limited period of time to allow for a migration. Follow a deprecated pattern and not just breaking. It would allow Batch Explorer-data template using`blobSource` to keep using it for a bit until most people migrated to latest BE. At this point template could be migrated to `httpUrl` and only break a small amount of users. Those would then either upgrade BE or could reference to a legacy branch using `blobSource`

File: azext/batch/models/_models_py3.py [hunk 1]
```diff
@@ -998,6 +1000,7 @@ def __init__(self,
             file_mode=file_mode,
             **kwargs)
         self.source = source
+        self.blob_source = blob_source
```
Comment (timotheeguerin) [inline azext/batch/models/_models_py3.py]: Could you add a deprecated annotation for this to it logs a python warning?

**Comment by xingwu1** [comment]
> I think it would be nice to support it for a limited period of time to allow for a migration. Follow a deprecated pattern and not just breaking. It would allow Batch Explorer-data template using`blobSource` to keep using it for a bit until most people migrated to latest BE. At this point template could be migrated to `httpUrl` and only break a small amount of users. Those would then either upgrade BE or could reference to a legacy branch using `blobSource`

Here is the plan, we will not support the blob_source in model, the SDK extension itself is breaking change, so you can't call from SDK and still use blob_source. However, we will keep blobSource back compatible at template level until next template version (?).

**Comment by bgklein** [comment]
Updated the flow a bit. Since we want this to be done only for templates moved to the level of handling json

**Comment by xingwu1** [comment]
@bgklein failed on pylint.

File: azext/batch/_template_utils.py [hunk 2]
```diff
@@ -1184,3 +1185,28 @@ def validate_json_object(json_obj, obj):
...
+    if 'blobSource' in resource_file:
+        if 'filePath' not in resource_file:
+            raise ValueError('Malformed ResourceFile: \'blobSource\' must '
+                             'also have \'file_path\' attribute')
+        resource_file['httpUrl'] = resource_file.pop('blobSource', None)
+        logger.warning('BlobSource has been updated to HttpUrl to reflect new '
+                       'functionality of accepting any http url instead of just'
+                       ' storage blobs. Please update your templates to'
+                       ' reflect this.')
+        return resource_file
```
Comment (xingwu1) [inline azext/batch/_template_utils.py]: this line is redundant.

**Pull request merged**